src/Security/RememberLoginHash.php

```diff
@@ -177,8 +177,17 @@ public function renew()
     public static function clear(Member $member, $alcDevice = null)
     {
         if (!$member->exists()) {
+            // If we don't have a valid user, we can't clear any "Remember me" tokens
             return;
         }
+
+        $logoutAcrossDevices = static::config()->logout_across_devices;
```

Comment:
This code is fine and works correctly, though read together with the block below with the existing logic, it's quite hard to follow it all (it's basically that existing code is hard to read)

Here's a proposed rewrite that makes everything a lot easier to read

```
        if (static::config()->logout_across_devices) {
            RememberLoginHash::get()->filter([
                'MemberID' => $member->ID
            ])->removeAll();
            return;
        }

        if ($alcDevice) {
            RememberLoginHash::get()->filter([
                'MemberID' => $member->ID,
                'DeviceID' => $alcDevice
            ])->removeAll();
        }
```

Reply:
Tweaked the logic as suggested.